### ISSUE TITLE
Installer can't download from the second cloudfront mirror

### ISSUE THREAD

**Issue by tombruijn** [appsignal/appsignal-elixir#683]
The installer was modified in #671 to support mirrors. When installer can't download from the first mirror (fastly) it will fall back on the second mirror (cloudfront).

When it tries to download from the second mirror hackney will return `{:error, :closed}` as the request response. This is not valid and will exit the installer.

The report doesn't detail that it failed downloading, only that hackney error listed above. With no clear error I first thought it failed the checksum verification.

## TODO

Ideas I have so far:

- [x] We need to improve the error reporting to report the actual download error. Look at the Ruby gem how it reports the error. https://github.com/appsignal/appsignal-elixir/pull/684
- [x] We need to figure out why the download from cloudfront fails. What is `{:error, :closed}` and how can we resolve it?
  - Could be an SSL issue. If I use `ssl_options: [verify: :verify_none]` (bad idea, never use this) it doesn't fail the download. So there's probably an SSL option we need to configure differently for cloudfront?
  - PR https://github.com/appsignal/appsignal-elixir/pull/688

## Environment details

```
$ elixir --version
Erlang/OTP 23 [erts-11.2.2.6] [source] [64-bit] [smp:2:2] [ds:2:2:10] [async-threads:1]

Elixir 1.11.4 (compiled with Erlang/OTP 23)
```

Packages (all latest at time of writing):

- hackney version: 1.18.0
- ssl_verify_fun: 1.1.6
- certifi version: 2.8.0

---

As reported in: https://app.intercom.com/a/apps/yzor8gyw/inbox/inbox/540709/conversations/16410700064173 (private link)
Example diagnose report from user: https://appsignal.com/admin/diagnose_reports/diagde902bd8a2572446c61b

**Comment by tombruijn** [comment]
User reports OTP (24.0.6) may resolve the issue. Requires further testing.

**Comment by tombruijn** [comment]
On OTP 24.1.2 I get the same error still 🤔 

```
$ cd /appsignal-elixir
$ rm -f /tmp/appsignal-agent-7376537.tar.gz && make clean && mix compile --no-deps-check
AppSignal installation failed: Could not download archive from any of our mirrors.
Please make sure your network allows access to any of these mirrors.
Attempted to download the archive from the following urls:
- <Other intentional error I forced to occur>

- URL: https://d135dj0rjqvssy.cloudfront.net/7376537/appsignal-x86_64-linux-all-static.tar.gz
- Error (hackney response):
{:error, :closed}
```

```
$ cat /usr/local/lib/erlang/releases/24/OTP_VERSION
24.1.2
```

**Comment by tombruijn** [comment]
PR #688 is merged. Now we need to release.

**Comment by tombruijn** [comment]
This issue has now been published in version 2.2.5.

@shairyar I think the issue was already resolved for the user who first encountered it, but you can let them know the issue has been fixed in the package if needed.
